app/stac_api/models.py

```diff
@@ -197,27 +190,469 @@ def update_geoadmin_variants(self, asset_geoadmin_variant, asset_proj_epsg, asse
+            elif new_end_datetime > self.cache_end_datetime:
+                # item's end_datetime did not define the right bound of
+                # the collection's temporal extent before update, which
+                # does not matter anyways, as it defines the right bound
+                # after update and collection's end_date can be simply
+                # adjusted
+                self.cache_end_datetime = new_end_datetime
+
+        self.save()
+
+    def delete_item_start_datetime(
+        self,
+        old_start_datetime,
+        new_start_datetime,
+        old_end_datetime,
+        new_end_datetime,
+        item_id,
+        qs=None
+    ):
+        '''
+        This function is called from within update_temporal_extent() when an
+        item is deleted from the collection to check if the collection's
+        start_datetime needs to be updated.
+        :param old_start_datetime: item's old value for properties_start_datetime
+        :param new_start_datetime: item's updated value for properties_start_datetime
+        :param old_end_datetime: item's old value for properties_end_datetime
+        :param new_end_datetime: item's updated value for properties_end_datetime
+        :param item_id: the id of the item being treated
+        :param qs: (optional) queryset with all items of the collection excluding the one
+        being updated.
+        '''
+        logger.debug(
+            'Deleting item %s from collection (adjusting start_datetime) %s',
+            item_id,
+            self.collection_name
+        )
+        if old_start_datetime == self.cache_start_datetime:
+            # item that is to be deleted defined left bound of collection's
+            # temporal extent
+            # first set cache_start_datetime to None, in case
+            # the currently deleted item is the only item of the collection
+            self.cache_start_datetime = None
+
+            logger.warning(
+                'Looping (twice) over all items of collection %s,'
+                'to update temporal extent, this may take a while',
+                self.collection_name
+            )
+
+            # get earliest start_datetime or none, in case none exists
+            if bool(qs.filter(properties_start_datetime__isnull=False)):
+                earliest_start_datetime = qs.filter(
+                    properties_start_datetime__isnull=False
+                ).earliest('properties_start_datetime').properties_start_datetime
+            else:
+                earliest_start_datetime = None
+
+            # get earliest datetime, or none in case none exists
+            if bool(qs.filter(properties_datetime__isnull=False)):
+                earliest_datetime = qs.filter(properties_datetime__isnull=False
+                                             ).earliest('properties_datetime').properties_datetime
+            else:
+                earliest_datetime = None
```

Comment:
Could use the same function as above ?

Reply:
see my comment above. Unfortunately, there are some important differences in the code (at least after my latest commits, fixing some small issues). So the function could not be reused elsewhere. And as the function, where the snippet is currently located, is quite short already, I'd leave it as it is, if OK with you.